apiserver/facades/controller/firewaller/firewaller.go

```diff
@@ -239,21 +239,7 @@ func (f *FirewallerAPIV3) GetMachinePorts(args params.MachinePortsParams) (param
 			result.Results[i].Error = common.ServerError(err)
 			continue
 		}
-		// TODO (hml) 2019-08-09
-		// Do we change the subnetTag from a Cidr to an ID?
-		// In the firewaller yes, for some of the juju cmds
-		// no?
-		var subnetID string
-		if subnetTag.Id() != "" {
-			// subnetTag.ID() is a CIDR not a subnet ID.
-			subnet, err := f.st.Subnet(subnetTag.Id())
-			if err != nil {
-				result.Results[i].Error = common.ServerError(err)
-				continue
-			}
-			subnetID = subnet.ID()
-		}
-		ports, err := machine.OpenedPorts(subnetID)
+		ports, err := machine.OpenedPorts(subnetTag.Id())
```

Comment:
How are we handling cross-version compatibility with this? (if we have an agent at 2.6 and a controller at 2.7, then the subnet tags are/aren't going to be what we want?)

Reply:
I think this is one is OK because the firewaller is a controller worker.

There do not appear to be any agents using subnet tags, but we will have to change params and bump facade versions so we can add subnets from the client with CIDR instead of `SubnetTag`.

We will also need to think about adding spaces from the client. That uses `SubnetTag`s which might be OK, but would be a documentation change for the CIDR-to-ID change. If we continue to allow CIDRs, a uniqueness check will be required.

`state.Subnet` now needs a rethink too. For the time-being it will be unique, but we are planning to change that.

Reply:
Yes, the firewater is a worker on the controller only.  As well, it turns out we don't actually set the subnet on any doc in OpenedPorts.  unit.OpenPorts() uses unit.OpenPortsOnSubnet with a blank subnet.  The same with unit.ClosePorts().  There is a lot of code around for the firewaller expecting a subnet to be used, but it always ends up as an empty string.  Looks like previous subnet work spotted short of getting that part implemented.  There is a 2016 TODO related to this in state/unit.go.

Reply:
The follow on work to this PR will be to not use a subnetTag in space and subnet client side commands.  The passed CIDR will require validation to replace removal of the SubnetTag.  Including incrementing some apiserver versions.

Reply:
So if we need a new Facade version, how do we handle the tags that would come in on the old version of the Facade and be passed in as a CIDR ?

Reply:
Assuming a new FirewallerAPIV6.  

First, FirewallerAPIV5.WatchOpenedPorts() would need a watcher in state to return CIDRs instead of SubnetIDs.  Otherwise the API won't be able to create the names.v2 SubnetTag. Creation of SubnetTags validates the contents are a CIDR.  Then FirewallerAPIV5.GetMachinePorts() would need to do translation of the CIDR to a SubnetID, as it was doing before this change. 

FirewallerAPIV5.GetMachineActiveSubnets() would need to return a string that looks like a names.v2 SubnetTag instead of the ID.  CIDR can be found with the ID.